harness/determined/cli/dev.py

```diff
@@ -348,6 +358,11 @@ def call_bindings(args: argparse.Namespace) -> None:
                         "call a function from bindings",
                         [
                             cli.Arg("name", help="name of the function to call"),
+                            cli.Arg(
+                                "--skip-auto-picking",
+                                help="skip auto-picking of only matching name on partial hits",
```

Comment:
Similar to the previous comment. I would reword this help text to say: "use matching function name if it is the only partial hit"

Reply:
would that still keep the same default behavior? I'm proposing for the default behavior to be what it is right now where it accepts the first match if it's the only match.

Reply:
Perhaps we have conflicting definitions of "auto-picking". Are you referring to auto-picking as bringing up the manual prompt? To me, if I see "auto-pick", I assume that means that the function itself is performing selection on behalf of the user somehow.

This function does one of 3 things:
1. call the binding function
2. if it finds one match in bindings, ask the user if that's what they're looking for an then continue
3. if it finds multiple matches, ask the user to disambiguate, then continue.

You want a flag that changes 2. Specifically, to skip asking the user if that's what they're looking for and instead continue with the regex'd binding function. It's essentially doing the same thing as `pip install -y`. Doing an auto-confirm. I want the flag to be some variation of "auto-confirm" and the description to also state something similar.

Reply:
> You want a flag that changes 2. Specifically, to skip asking the user if that's what they're looking for and instead continue with the regex'd binding function. It's essentially doing the same thing as pip install -y

yes!
(before this pr points 2 and 3 are not differentiated at all)

> I want the flag to be some variation of "auto-confirm" and the description to also state something similar.

yeah auto confirm is a good way of putting it, that works for me. 

are you also asking that we don't default to auto confirming if there's only a single match? because my point with this pr is to make it the default but also add an option to get the old behavior if anyone wanted it

Reply:
Yeah I'd rather have it not auto-confirm by default and require an `--auto-confirm` flag. It makes more sense to err on the side of caution and only add anoption for users that are absolutely sure that's what they want.

Reply:
yeah, that would generally be safer. It's hard to think through all the ways this could potentially lead the users to make the wrong API call 🤔 maybe we do go with the safer default and we'll keep using it with the flag for now and that'll give us a better idea. I could think of more complicated ways of trying to estimate the risks of auto confirming for the user, eg `get_*` calls can be assumed read-only and safe but we should probably keep it simple